module/builder/consensus/builder.go

```diff
@@ -445,20 +464,35 @@ func (b *Builder) getInsertableReceipts(parentID flow.Identifier) ([]*flow.Execu
 		return nil, fmt.Errorf("failed to retrieve reachable receipts from memool: %w", err)
 	}
 
+	filteredReceipts := make([]*flow.ExecutionReceiptMeta, 0, len(receipts))
+	results := make([]*flow.ExecutionResult, 0)
+	for _, receipt := range receipts {
+		resultID := receipt.ExecutionResult.ID()
+		if _, inserted := includedResults[resultID]; !inserted {
+			results = append(results, &receipt.ExecutionResult)
+			includedResults[resultID] = struct{}{}
+		}
+
+		filteredReceipts = append(filteredReceipts, receipt.Meta())
```

Comment:
1We are computing the `ExecutionResult.ID()` twice, since `receipt.Meta()` also computes it. So we could first run `receipt.Meta()`, and then read the resultID from the Meta